homeassistant/components/vallox/__init__.py

```diff
@@ -171,7 +173,15 @@ async def async_update_data() -> ValloxState:
     hass.data[DOMAIN] = {"client": client, "coordinator": coordinator, "name": name}
 
     async def _async_load_platform_delayed(*_: Any) -> None:
-        await coordinator.async_refresh()
+        # We need a successful update before loading the platforms, because platform init code
+        # derives the UUIDs from the data the coordinator fetches.
```

Comment:
Why not use `await coordinator.async_config_entry_first_refresh()` the first time?

Reply:
Please see https://github.com/home-assistant/core/pull/62227/files/6aa9374a2f9f6eb2d1b70dde1f708331c46e94d3#diff-eb4c41645a12264b6231264fdd746a6f40691b623471cec7c5f72a0981083ac8 where this was the first attempt.

Reply:
Right, so why not convert it to use config entries instead in that case?

Reply:
Due to lack of time resources on my side. Combined with multiple people out there for which the Vallox integration is broken right now and are waiting for a quick fix. I would also deem it risky to upgrade the whole thing to ConfigEntry for hotfixing this in between two releases?

Unfortunately, we cannot just revert the commit that did the damage (5cc594682f216250adb635f478886c2058bef1be), because then the other part of the user base would end up with this situation:
https://github.com/home-assistant/core/issues/61708#issuecomment-997369036

Reply:
Upgrading to config entries should of course be the next goal to fix this idiomatically, I agree.